File: examples/vulnerable-test.py
"""
Test file with 15+ vulnerabilities
"""
import os
import pickle
import subprocess
import hashlib

# 1. Hardcoded secret (CRITICAL)
API_KEY = "sk-1234567890abcdefghijklmnopqrstuvwxyz"
PASSWORD = "MySecretPassword123"

# 2. SQL Injection (CRITICAL)
def get_user(user_id):
    query = f"SELECT * FROM users WHERE id = '{user_id}'"
    return db.execute(query)

# 3. Command Injection (HIGH)
def run_command(cmd):
    os.system(cmd)
    subprocess.call(cmd, shell=True)

# 4. Weak Crypto (MEDIUM)
def hash_password(pwd):
    return hashlib.md5(pwd.encode()).hexdigest()

# 5. Insecure Deserialization (MEDIUM)
def load_data(data):
    return pickle.loads(data)

# 6. Eval (HIGH)
def calculate(formula):
    return eval(formula)

print("Test file with multiple vulnerabilities")

File: examples/vulnerable-test.js
/**
 * Test file with 10+ vulnerabilities
 */

// 1. Hardcoded secret (CRITICAL)
const API_KEY = "sk-1234567890abcdefghijklmnopqrstuvwxyz";
const DB_PASSWORD = "admin123";

// 2. XSS (HIGH)
function displayInput(input) {
    document.getElementById('output').innerHTML = input;
}

// 3. Eval (HIGH)
function calculate(formula) {
    return eval(formula);
}

// 4. Weak crypto (MEDIUM)
const crypto = require('crypto');
function hashPassword(pwd) {
    return crypto.createHash('md5').update(pwd).digest('hex');
}

// 5. Command injection (HIGH)
const { exec } = require('child_process');
function runCommand(cmd) {
    exec(`ls ${cmd}`);
}

console.log("Test file with vulnerabilities");
